src/notification-banner/useNotificationBanners.jsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Alert from 'terra-alert';
+import Button from 'terra-button';
+import BannerRegistrationContext from './private/BannerRegistrationContext';
+
+import { organizeBannersByPriority } from './private/utils';
+
+/**
+ * The `useNotificationBanners` Hook manages registering and prioritizing Notification Banners
+ * rendered within the Notification Banner Context.
+ *
+ * Returns:
+ *   - NotificationBannerProvider - React Context Provider - Provides the Banner Registration Context to its children.
+ *         This allows any NotificationBanner registered beneath it to be displayed in teh NotificationBanner's list.
+ *   - NotificationBanners - React Component - Renders a list of prioritized notification banners.
+ *
+ * @returns { NotificationBannerProvider, NotificationBanners }
+ */
+const useNotificationBanners = () => {
+  /** The registeredBanners ref maintains collection of banner ids and banner props registered to the  NotificationBannerProvider. */
+  const registeredBanners = React.useRef({});
+
+  /**
+   * The updateBannerState ref stores the update state function used to manage the banner rendered in the NotificationBanners component.
+   * This ties the state updates to the `useNotificationBanners` hook, while allowing the NotificationBanners to be rendered above or below
+   * the NotificationBannerProvider.
+   */
+  const updateBannerState = React.useRef();
```

Comment:
@tbiethman I went forward with this approach. It seems to work great and have yet to notice any unexpected consequences. Add a bunch of docs to hopefully explain what is going on. Will see what other think.